File: library/json-mapper2/src/main/scala/com/anjunar/scala/schema/model/CollectionDescriptor.scala
package com.anjunar.scala.schema.model

import com.anjunar.scala.mapper.annotations.IgnoreFilter

import scala.beans.BeanProperty
import scala.compiletime.uninitialized

@IgnoreFilter
class CollectionDescriptor extends NodeDescriptor {
  
  @BeanProperty
  var items : NodeDescriptor = uninitialized
  
}

File: application/src/main/typescript/packages/library/src/domain/descriptors/CollectionDescriptor.ts
import NodeDescriptor from "./NodeDescriptor";
import Basic from "../../mapper/annotations/Basic";
import Entity from "../../mapper/annotations/Entity";
import ObjectDescriptor from "./ObjectDescriptor";

@Entity("collectionDescriptor")
export default class CollectionDescriptor extends NodeDescriptor    {

    $type = "collectionDescriptor"

    @Basic()
    items : ObjectDescriptor

}